integration_test/cases/interval.exs

```diff
@@ -235,6 +244,19 @@ defmodule Ecto.Integration.IntervalTest do
            TestRepo.all(from p in Post, where: field(p, ^field) > datetime_add(^inserted_at, -5, ^"day"))
   end
 
+  test "datetime_add with Ecto.DateTime" do
+    inserted_at = @inserted_at |> NaiveDateTime.to_erl |> Ecto.DateTime.from_erl
+    field = :inserted_at
+    assert [_]  =
+      TestRepo.all(from p in Post, where: p.inserted_at > datetime_add(^inserted_at, ^-1, "year"))
+    assert [_]  =
+      TestRepo.all(from p in Post, where: p.inserted_at > datetime_add(^inserted_at, -3, "month"))
+    assert [_]  =
+      TestRepo.all(from p in Post, where: field(p, ^field) > datetime_add(^inserted_at, ^-3, ^"week"))
```

Comment:
@josevalim I am guessing this is warning on `^` or `-`. Should this be reported to credo?